src/wasm/wasm-stack.cpp

```diff
@@ -1859,22 +1859,33 @@ void BinaryInstWriter::visitTry(Try* curr) {
   emitResultType(curr->type);
 }
 
-void BinaryInstWriter::emitCatch(Try* curr) {
+void BinaryInstWriter::emitCatch(Try* curr, Index i) {
   assert(!breakStack.empty());
   breakStack.pop_back();
   breakStack.emplace_back(IMPOSSIBLE_CONTINUE);
   if (func && !sourceMap) {
     parent.writeExtraDebugLocation(curr, func, BinaryLocations::Catch);
   }
-  o << int8_t(BinaryConsts::Catch);
+  o << int8_t(BinaryConsts::Catch)
+    << U32LEB(parent.getEventIndex(curr->catchEvents[i]));
+}
+
+void BinaryInstWriter::emitCatchAll(Try* curr) {
+  assert(!breakStack.empty());
+  breakStack.pop_back();
+  breakStack.emplace_back(IMPOSSIBLE_CONTINUE);
+  if (func && !sourceMap) {
+    parent.writeExtraDebugLocation(curr, func, BinaryLocations::Catch);
+  }
```

Comment:
I don't know much about debugging support, so I just copied the code for `emitCatch`, but is this right? Should we add `BinaryLocations::CatchAll` or something? Is it correct to print this info per `catch` clause, when there are multiple `catch`/`catch_all` clauses per `try`?

Reply:
Hmm, I think there is indeed a problem here. The current model of `writeExtraDebugLocation` is that an instruction has one or two more special locations, like an else or a catch. So each call with `::Catch` here overwrites the previous one.

I think we'll need to make `DelimiterLocations` an array. However, that would be nontrivial for a few reasons. For this PR, maybe just write TODOs for debug info for try-catch?

Reply:
Done